[Kommander 2.2] Provide instructions to detach < 2.2 metallb from kommander (#4279)
mesosphere/dcos-docs-site: pages/dkp/kommander/2.2/dkp-upgrade/upgrade-kommander/index.md

-   Review the [Platform Application version updates][release_notes] that are part of this upgrade.  
-   For air-gapped environments **with** DKP Catalog Applications in a multi-cluster environment: [Load the Docker images into your Docker registry][load_images_catalog]
-   For air-gapped environments **without** DKP Catalog Applications: [Load the Docker images into your Docker registry][load_images]

  Download the Kommander application definitions:

  ```bash
  wget "https://downloads.d2iq.com/dkp/v2.2.0/kommander-applications-v2.2.0.tar.gz"
  ```

  Download the Kommander charts bundle:

  ```bash
  wget "https://downloads.d2iq.com/dkp/v2.2.0/dkp-kommander-charts-bundle-v2.2.0.tar.gz"
  ```

  If you have any DKP Catalog Applications, download the DKP Catalog Application charts bundle:

  ```bash
  wget "https://downloads.d2iq.com/dkp/v2.2.0/dkp-catalog-applications-charts-bundle-v2.2.0.tar.gz"
  ```
  

>   <p class="message--note"><strong>NOTE:</strong> Beginning with DKP version 2.2, MetalLB is no longer managed as a catalog application. If you installed MetalLB on the cluster that you're upgrading prior to DKP version 2.2, you will need to detach MetalLB from the cluster prior to upgrading.</p>

Reviewer: ```suggestion
  <p class="message--important"><strong>NOTE:</strong> Beginning with DKP version 2.2, MetalLB is no longer managed as a platform application. If you installed MetalLB on the cluster that you're upgrading prior to DKP version 2.2, you will need to detach MetalLB from the cluster prior to upgrading.</p>
```

Might bump this one up to an important

Reviewer: Maybe we can include this section under a subheading to further highlight these steps?

>   <p class="message--note"><strong>NOTE:</strong> Beginning with DKP version 2.2, MetalLB is no longer managed as a catalog application. If you installed MetalLB on     the cluster that you're upgrading prior to DKP version 2.2, you will need to detach MetalLB from the cluster prior to upgrading..</p>

Reviewer 2: ```suggestion
  <p class="message--note"><strong>NOTE:</strong> Beginning with DKP version 2.2, MetalLB is no longer managed as a catalog application. If you installed MetalLB on the cluster that you're upgrading prior to DKP version 2.2, you will need to detach MetalLB from the cluster prior to upgrading.</p>
```

  
  1. Pause the helm release.

>   ```sh

Reviewer 2: I think all of these should be bash? but not sure... 
```suggestion
  ```bash
```

  kubectl -n kommander patch -p='{"spec":{"suspend": true}}' --type=merge helmrelease/metallb

>   helmrelease.helm.toolkit.fluxcd.io/metallb patched

Reviewer: this line is technically the output, so it should go in a separate ```sh section

  ```

  1. Delete the helm release secret.
  ```bash
  kubectl -n kommander delete secret -l name=metallb,owner=helm

>   secret "sh.helm.release.v1.metallb.v1" deleted

Reviewer: same here, this is output

  ```
  1. Delete MetalLB.
  ```bash
  k -n kommander delete appdeployment metallb

>   appdeployment.apps.kommander.d2iq.io "metallb" deleted

Reviewer: same here, this is output

  ```

  1. Unpause the helm release.
  ```bash
  kubectl -n kommander patch -p='{"spec":{"suspend": false}}' --type=merge helmrelease/metallb

>   helmrelease.helm.toolkit.fluxcd.io/metallb patched

Reviewer: same here, this is output

  ```
  This deletes MetalLb from Kommander while leaving the resources running in the cluster.
  

>   ```bash
>   $ kubectl -n kommander get pod -l app=metallb

Reviewer: ```suggestion
  ```bash
  kubectl -n kommander get pod -l app=metallb
```

maybe separate out the command + sample output?

>   NAME                                 READY   STATUS    RESTARTS   AGE

Reviewer: ```suggestion
  
  ```sh
  NAME                                 READY   STATUS    RESTARTS   AGE
```

too many ticks to suggest this properly 🤦🏻‍♀️  but essentially wrapping the kubectl cmd in bash
and then separating out the output into sh

-   Review the [Platform Application version updates][release_notes] that are part of this upgrade.  
-   For air-gapped environments **with** DKP Catalog Applications in a multi-cluster environment: [Load the Docker images into your Docker registry][load_images_catalog]
-   For air-gapped environments **without** DKP Catalog Applications: [Load the Docker images into your Docker registry][load_images]

  Download the Kommander application definitions:

  ```bash
  wget "https://downloads.d2iq.com/dkp/v2.2.0/kommander-applications-v2.2.0.tar.gz"
  ```

  Download the Kommander charts bundle:

  ```bash
  wget "https://downloads.d2iq.com/dkp/v2.2.0/dkp-kommander-charts-bundle-v2.2.0.tar.gz"
  ```

  If you have any DKP Catalog Applications, download the DKP Catalog Application charts bundle:

  ```bash
  wget "https://downloads.d2iq.com/dkp/v2.2.0/dkp-catalog-applications-charts-bundle-v2.2.0.tar.gz"
  ```

> ## Detatch MetalLB from Kommander

Reviewer: ```suggestion
## Detach MetalLB from Kommander
```

  <p class="message--important"><strong>NOTE:</strong> Beginning with DKP version 2.2, MetalLB is no longer managed as a platform application. If you installed MetalLB on the cluster that you're upgrading prior to DKP version 2.2, you will need to detach MetalLB from the cluster prior to upgrading.</p>
  
  1. Pause the helm release.
  ```bash

>   kubectl -n kommander patch -p='{"spec":{"suspend": true}}' --type=merge helmrelease/metallb

Reviewer: missed a spot to separate out the input/outputs here!

  helmrelease.helm.toolkit.fluxcd.io/metallb patched
  ```

  1. Delete the helm release secret.
  ```bash
  kubectl -n kommander delete secret -l name=metallb,owner=helm
  ```
  ```sh 
  secret "sh.helm.release.v1.metallb.v1" deleted
  ```
  
  1. Delete MetalLB.
  ```bash
  k -n kommander delete appdeployment metallb
  ```
  
  ```sh 
  appdeployment.apps.kommander.d2iq.io "metallb" deleted
  ```

  1. Unpause the helm release.
  ```bash
  kubectl -n kommander patch -p='{"spec":{"suspend": false}}' --type=merge helmrelease/metallb
  ```
  
  ```sh 
  helmrelease.helm.toolkit.fluxcd.io/metallb patched
  ```
  This deletes MetalLb from Kommander while leaving the resources running in the cluster.
  
  ```bash

>   kubectl -n kommander get pod -l app=metallb

Reviewer: can't suggest ticks properly 🙈 but there needs to be a closing ``` here
